### ISSUE TITLE
failed to query event: CUDA_ERROR_LAUNCH_FAILED

### ISSUE THREAD

**Issue by monikatomar92** [tensorflow/tensorflow#22477]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: 
Linux Ubuntu 18.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
NA
- **TensorFlow installed from (source or binary)**: 
binary
- **TensorFlow version (use command below)**:
1.10.1
- **Python version**:
3.6.5
- **Bazel version (if compiling from source)**:
NA
- **GCC/Compiler version (if compiling from source)**:
NA
- **CUDA/cuDNN version**:
9.0 cuDNN 7.2.1
- **GPU model and memory**:
GeForce GTX 1080 Ti 11GB
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Encountered this error while running a image classification training script. 
E tensorflow/stream_executor/cuda/cuda_event.cc:48] Error polling for event status: failed to query event: CUDA_ERROR_LAUNCH_FAILED
2018-09-23 18:31:38.983015: F tensorflow/core/common_runtime/gpu/gpu_event_mgr.cc:206] Unexpected Event status: 1
Aborted (core dumped)

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by wt-huang** [comment]
@monikatomar92 Could you post your tf_env.txt here after environment capture script?

**Comment by monikatomar92** [comment]
Hi. Thank you, Following is the tf_env.txt
[tf_env.txt](https://github.com/tensorflow/tensorflow/files/2413448/tf_env.txt)

**Comment by wt-huang** [comment]
@monikatomar92 You may need to install CUDA9.2. Could you post the files under /usr/local/cuda-9.0/lib64?

**Comment by monikatomar92** [comment]
These are the files I have under /usr/local/cuda-9.0/lib64
libaccinj64.so          libcusolver_static.a    libnppist_static.a
libaccinj64.so.9.0      libcusparse.so          libnppisu.so
libaccinj64.so.9.0.176  libcusparse.so.9.0      libnppisu.so.9.0
libcublas_device.a      libcusparse.so.9.0.176  libnppisu.so.9.0.176
libcublas.so            libcusparse_static.a    libnppisu_static.a
libcublas.so.9.0        libnppc.so              libnppitc.so
libcublas.so.9.0.176    libnppc.so.9.0          libnppitc.so.9.0
libcublas.so.9.0.282    libnppc.so.9.0.176      libnppitc.so.9.0.176
libcublas.so.9.0.333    libnppc_static.a        libnppitc_static.a
libcublas.so.9.0.425    libnppial.so            libnpps.so
libcublas.so.9.0.480    libnppial.so.9.0        libnpps.so.9.0
libcublas_static.a      libnppial.so.9.0.176    libnpps.so.9.0.176
libcudadevrt.a          libnppial_static.a      libnpps_static.a
libcudart.so            libnppicc.so            libnvblas.so
libcudart.so.9.0        libnppicc.so.9.0        libnvblas.so.9.0
libcudart.so.9.0.176    libnppicc.so.9.0.176    libnvblas.so.9.0.176
libcudart_static.a      libnppicc_static.a      libnvblas.so.9.0.282
libcudnn.so             libnppicom.so           libnvblas.so.9.0.333
libcudnn.so.7           libnppicom.so.9.0       libnvblas.so.9.0.425
libcudnn.so.7.2.1       libnppicom.so.9.0.176   libnvblas.so.9.0.480
libcudnn_static.a       libnppicom_static.a     libnvgraph.so
libcufft.so             libnppidei.so           libnvgraph.so.9.0
libcufft.so.9.0         libnppidei.so.9.0       libnvgraph.so.9.0.176
libcufft.so.9.0.176     libnppidei.so.9.0.176   libnvgraph_static.a
libcufft_static.a       libnppidei_static.a     libnvrtc-builtins.so
libcufftw.so            libnppif.so             libnvrtc-builtins.so.9.0
libcufftw.so.9.0        libnppif.so.9.0         libnvrtc-builtins.so.9.0.176
libcufftw.so.9.0.176    libnppif.so.9.0.176     libnvrtc.so
libcufftw_static.a      libnppif_static.a       libnvrtc.so.9.0
libcuinj64.so           libnppig.so             libnvrtc.so.9.0.176
libcuinj64.so.9.0       libnppig.so.9.0         libnvToolsExt.so
libcuinj64.so.9.0.176   libnppig.so.9.0.176     libnvToolsExt.so.1
libculibos.a            libnppig_static.a       libnvToolsExt.so.1.0.0
libcurand.so            libnppim.so             libOpenCL.so
libcurand.so.9.0        libnppim.so.9.0         libOpenCL.so.1
libcurand.so.9.0.176    libnppim.so.9.0.176     libOpenCL.so.1.0
libcurand_static.a      libnppim_static.a       libOpenCL.so.1.0.0
libcusolver.so          libnppist.so            stubs
libcusolver.so.9.0      libnppist.so.9.0
libcusolver.so.9.0.176  libnppist.so.9.0.176

**Comment by wt-huang** [comment]
@monikatomar92 It appears that you need to install CUDA9.2 for your GPU. Also make sure that `LD_LIBRARY_PATH` is set up correctly.

**Comment by monikatomar92** [comment]
@wt-huang I installed CUDA9.2 but I am not able to install/build tensorflow for the same. Any suggestions with if the latest release of tensorflow would work with CUDA9.2?

**Comment by wt-huang** [comment]
@monikatomar92 Please post the errors here.

**Comment by monikatomar92** [comment]
ERROR: /home/monika/tensorflow/tensorflow/core/BUILD:1010:1: C++ compilation of rule '//tensorflow/core:logging_ops_op_lib' failed (Exit 1): crosstool_wrapper_driver_is_not_gcc failed: error executing command 
  (cd /home/monika/.cache/bazel/_bazel_monika/c1ba60c7350215d6c41b5c2e155ce675/execroot/org_tensorflow && \
  exec env - \
    LD_LIBRARY_PATH=:/usr/local/cuda/lib64 \
    PATH=/home/monika/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/monika/bin:/usr/local/cuda/bin:/home/monika/bin:/home/monika/bin \
    PWD=/proc/self/cwd \
  external/local_config_cuda/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc -MD -MF bazel-out/host/bin/tensorflow/core/_objs/logging_ops_op_lib/logging_ops.pic.d '-frandom-seed=bazel-out/host/bin/tensorflow/core/_objs/logging_ops_op_lib/logging_ops.pic.o' -DEIGEN_MPL2_ONLY '-DEIGEN_MAX_ALIGN_BYTES=64' -D__CLANG_SUPPORT_DYN_ANNOTATION__ -DTF_USE_SNAPPY -iquote . -iquote bazel-out/host/genfiles -iquote bazel-out/host/bin -iquote external/bazel_tools -iquote bazel-out/host/genfiles/external/bazel_tools -iquote bazel-out/host/bin/external/bazel_tools -iquote external/eigen_archive -iquote bazel-out/host/genfiles/external/eigen_archive -iquote bazel-out/host/bin/external/eigen_archive -iquote external/local_config_sycl -iquote bazel-out/host/genfiles/external/local_config_sycl -iquote bazel-out/host/bin/external/local_config_sycl -iquote external/com_google_absl -iquote bazel-out/host/genfiles/external/com_google_absl -iquote bazel-out/host/bin/external/com_google_absl -iquote external/nsync -iquote bazel-out/host/genfiles/external/nsync -iquote bazel-out/host/bin/external/nsync -iquote external/gif_archive -iquote bazel-out/host/genfiles/external/gif_archive -iquote bazel-out/host/bin/external/gif_archive -iquote external/jpeg -iquote bazel-out/host/genfiles/external/jpeg -iquote bazel-out/host/bin/external/jpeg -iquote external/protobuf_archive -iquote bazel-out/host/genfiles/external/protobuf_archive -iquote bazel-out/host/bin/external/protobuf_archive -iquote external/com_googlesource_code_re2 -iquote bazel-out/host/genfiles/external/com_googlesource_code_re2 -iquote bazel-out/host/bin/external/com_googlesource_code_re2 -iquote external/farmhash_archive -iquote bazel-out/host/genfiles/external/farmhash_archive -iquote bazel-out/host/bin/external/farmhash_archive -iquote external/fft2d -iquote bazel-out/host/genfiles/external/fft2d -iquote bazel-out/host/bin/external/fft2d -iquote external/highwayhash -iquote bazel-out/host/genfiles/external/highwayhash -iquote bazel-out/host/bin/external/highwayhash -iquote external/zlib_archive -iquote bazel-out/host/genfiles/external/zlib_archive -iquote bazel-out/host/bin/external/zlib_archive -iquote external/local_config_cuda -iquote bazel-out/host/genfiles/external/local_config_cuda -iquote bazel-out/host/bin/external/local_config_cuda -isystem external/com_google_absl -isystem bazel-out/host/genfiles/external/com_google_absl -isystem bazel-out/host/bin/external/com_google_absl -isystem external/eigen_archive -isystem bazel-out/host/genfiles/external/eigen_archive -isystem bazel-out/host/bin/external/eigen_archive -isystem external/nsync/public -isystem bazel-out/host/genfiles/external/nsync/public -isystem bazel-out/host/bin/external/nsync/public -isystem external/gif_archive/lib -isystem bazel-out/host/genfiles/external/gif_archive/lib -isystem bazel-out/host/bin/external/gif_archive/lib -isystem external/protobuf_archive/src -isystem bazel-out/host/genfiles/external/protobuf_archive/src -isystem bazel-out/host/bin/external/protobuf_archive/src -isystem external/farmhash_archive/src -isystem bazel-out/host/genfiles/external/farmhash_archive/src -isystem bazel-out/host/bin/external/farmhash_archive/src -isystem external/zlib_archive -isystem bazel-out/host/genfiles/external/zlib_archive -isystem bazel-out/host/bin/external/zlib_archive -isystem external/local_config_cuda/cuda -isystem bazel-out/host/genfiles/external/local_config_cuda/cuda -isystem bazel-out/host/bin/external/local_config_cuda/cuda -isystem external/local_config_cuda/cuda/cuda/include -isystem bazel-out/host/genfiles/external/local_config_cuda/cuda/cuda/include -isystem bazel-out/host/bin/external/local_config_cuda/cuda/cuda/include -isystem external/local_config_cuda/cuda/cuda/include/crt -isystem bazel-out/host/genfiles/external/local_config_cuda/cuda/cuda/include/crt -isystem bazel-out/host/bin/external/local_config_cuda/cuda/cuda/include/crt '-std=c++11' -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -fPIC -U_FORTIFY_SOURCE '-D_FORTIFY_SOURCE=1' -fstack-protector -Wall -fno-omit-frame-pointer -no-canonical-prefixes -DNDEBUG -g0 -O2 -ffunction-sections -fdata-sections -g0 '-march=native' -g0 -DEIGEN_AVOID_STL_ARRAY -Iexternal/gemmlowp -Wno-sign-compare -fno-exceptions '-ftemplate-depth=900' '-DGOOGLE_CUDA=1' -msse3 -pthread -c tensorflow/core/ops/logging_ops.cc -o bazel-out/host/bin/tensorflow/core/_objs/logging_ops_op_lib/logging_ops.pic.o)
In file included from external/protobuf_archive/src/google/protobuf/wire_format_lite_inl.h:44:0,
                 from external/protobuf_archive/src/google/protobuf/map_type_handler.h:35,
                 from external/protobuf_archive/src/google/protobuf/map.h:49,
                 from external/protobuf_archive/src/google/protobuf/generated_message_table_driven.h:34,
                 from bazel-out/host/genfiles/tensorflow/core/framework/types.pb.h:25,
                 from ./tensorflow/core/framework/tensor_shape.h:22,
                 from ./tensorflow/core/framework/partial_tensor_shape.h:20,
                 from ./tensorflow/core/framework/attr_value_util.h:23,
                 from ./tensorflow/core/framework/node_def_util.h:23,
                 from ./tensorflow/core/framework/shape_inference.h:20,
                 from ./tensorflow/core/framework/common_shape_fns.h:20,
                 from tensorflow/core/ops/logging_ops.cc:16:
external/protobuf_archive/src/google/protobuf/repeated_field.h: In instantiation of 'typename TypeHandler::Type* google::protobuf::internal::RepeatedPtrFieldBase::Add(typename TypeHandler::Type*) [with TypeHandler = google::protobuf::RepeatedPtrField<tensorflow::TensorShapeProto>::TypeHandler; typename TypeHandler::Type = tensorflow::TensorShapeProto]':
external/protobuf_archive/src/google/protobuf/repeated_field.h:1977:49:   required from 'Element* google::protobuf::RepeatedPtrField<T>::Add() [with Element = tensorflow::TensorShapeProto]'
bazel-out/host/genfiles/tensorflow/core/framework/attr_value.pb.h:1001:21:   required from here
external/protobuf_archive/src/google/protobuf/repeated_field.h:1539:1: internal compiler error: Segmentation fault
 }
 ^
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-4.8/README.Bugs> for instructions.
The bug is not reproducible, so it is likely a hardware or OS problem.
Target //tensorflow/tools/pip_package:build_pip_package failed to build
INFO: Elapsed time: 139.522s, Critical Path: 57.15s
INFO: 221 processes: 221 local.
FAILED: Build did NOT complete successfully

**Comment by monikatomar92** [comment]
@wt-huang I tried setting up this environment as you mentioned in the last comment. I still couldn't build the tensorflow. Got the following error :
ERROR: /home/monika/.cache/bazel/_bazel_monika/c1ba60c7350215d6c41b5c2e155ce675/external/nasm/BUILD.bazel:8:1: C++ compilation of rule '@nasm//:nasm' failed (Exit 1): crosstool_wrapper_driver_is_not_gcc failed: error executing command 
  (cd /home/monika/.cache/bazel/_bazel_monika/c1ba60c7350215d6c41b5c2e155ce675/execroot/org_tensorflow && \
  exec env - \
    LD_LIBRARY_PATH=:/usr/local/cuda/lib64:/usr/local/cuda/lib64 \
    PATH=/home/monika/anaconda3/bin:/home/monika/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/monika/bin:/usr/local/cuda/bin:/home/monika/bin:/home/monika/bin:/home/monika/bin:/usr/local/cuda/bin:/home/monika/bin:/home/monika/bin \
    PWD=/proc/self/cwd \
  external/local_config_cuda/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc -MD -MF bazel-out/host/bin/external/nasm/_objs/nasm/external/nasm/asm/nasm.d -DHAVE_SNPRINTF -DHAVE_SYS_TYPES_H -iquote external/nasm -iquote bazel-out/host/genfiles/external/nasm -iquote external/bazel_tools -iquote bazel-out/host/genfiles/external/bazel_tools -isystem external/nasm/asm -isystem bazel-out/host/genfiles/external/nasm/asm -isystem bazel-out/host/bin/external/nasm/asm -isystem external/nasm/include -isystem bazel-out/host/genfiles/external/nasm/include -isystem bazel-out/host/bin/external/nasm/include -isystem external/nasm/output -isystem bazel-out/host/genfiles/external/nasm/output -isystem bazel-out/host/bin/external/nasm/output -isystem external/nasm/x86 -isystem bazel-out/host/genfiles/external/nasm/x86 -isystem bazel-out/host/bin/external/nasm/x86 -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -fPIE -U_FORTIFY_SOURCE '-D_FORTIFY_SOURCE=1' -fstack-protector -Wall -fno-omit-frame-pointer -no-canonical-prefixes -DNDEBUG -g0 -O2 -ffunction-sections -fdata-sections -g0 '-march=native' -w '-std=c99' -c external/nasm/asm/nasm.c -o bazel-out/host/bin/external/nasm/_objs/nasm/external/nasm/asm/nasm.o)
external/nasm/asm/nasm.c: In function ‘main’:
external/nasm/asm/nasm.c:517:1: internal compiler error: in hash_rtx_cb, at cse.c:2533
 }
 ^
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-4.8/README.Bugs> for instructions.
The bug is not reproducible, so it is likely a hardware or OS problem.
Target //tensorflow/tools/pip_package:build_pip_package failed to build
INFO: Elapsed time: 25.844s, Critical Path: 1.54s
INFO: 94 processes: 94 local.
FAILED: Build did NOT complete successfully

**Comment by wt-huang** [comment]
@monikatomar92 I was able to install TensorFlow using the said environment. It seems that crosstool/gcc is not installed correctly in your system. You can either follow [this](https://medium.com/@Oysiyl/install-tensorflow-1-8-0-with-gpu-from-source-on-ubuntu-18-04-bionic-beaver-35cfa9df3600) to install TensorFlow or use [docker](https://www.tensorflow.org/install/docker).

**Comment by wt-huang** [comment]
Closing this issue for now, feel free to reopen if problem persists.
